ambry-messageformat/src/main/java/com/github/ambry/messageformat/BlobPropertiesSerDe.java

```diff
@@ -60,22 +62,22 @@ public static BlobProperties getBlobPropertiesFromStream(DataInputStream stream)
     short accountId = version > VERSION_1 ? stream.readShort() : Account.UNKNOWN_ACCOUNT_ID;
     short containerId = version > VERSION_1 ? stream.readShort() : Container.UNKNOWN_CONTAINER_ID;
     boolean isEncrypted = version > VERSION_2 && stream.readByte() == (byte) 1;
-    String contentEncoding = version > VERSION_3 ? Utils.readIntString(stream) : null;
-    String filename = version > VERSION_3 ? Utils.readIntString(stream) : null;
+    String contentEncoding = version > VERSION_3 ? Utils.readNullableIntString(stream) : null;
+    String filename = version > VERSION_3 ? Utils.readNullableIntString(stream) : null;
```

Comment:
minor: looks like we don't have unit test that version = 4 but either contentEncoding or filename is null.

Reply:
I've added an unit test in BlobPropertyTest which set contentEncoding and filename is null.